src/core/util.cpp

```diff
@@ -643,142 +651,363 @@ namespace mamba
         return prepend(p.c_str(), start, newline);
     }
 
-    int try_lock(int fd, int pid, bool wait)
+    LockFile::LockFile(const fs::path& path, const std::chrono::seconds& timeout)
+        : m_path(path)
+        , m_timeout(timeout)
     {
-        // very much inspired by boost file_lock and
-        // dnf's https://github.com/rpm-software-management/dnf lock.py#L80
-#ifdef _WIN32
-        int ret;
-        if (!wait)
+        m_fd = open(path.c_str(), O_RDWR | O_CREAT, 0666);
+    }
+
+    LockFile::~LockFile()
+    {
+        if (m_fd > -1)
+            close_fd();
+    }
+
+    void LockFile::remove() noexcept
+    {
+        close_fd();
+
+        std::error_code ec;
+        LOG_TRACE << "Removing file '" << m_path.string() << "'";
+        fs::remove(m_path, ec);
+
+        if (ec)
         {
-            ret = _locking(fd, LK_NBLCK, 1 /*lock_file_contents_length()*/);
+            LOG_ERROR << "Removing lock file '" << m_path.string() << "' failed\n"
```

Comment:
Just printing path should automatically `quote` it. Don't need to call `string` on it.

Reply:
yeah I saw your comment, I would prefer to follow what's done everywhere else, which is currently printing `'/my/path'` rather than having both `'/my/path'` and `"/my/path"`